React redux prototype

kusera opened the pull request:
- [x] Create Redux prototype
- [x] Create a component for the header
- [x] Create a component for the control group label
- [x] Create a component for filter buttons
- [x] Create a component for the faction switch
- [x] Create a component for the footer
- [x] Create a static component for the map
- [x] Hook up actions to react components

Commit:
Fix syntax errors and add details

Commit:
Change indent from 4 spaces to 2

Commit:
Add google fonts, halve tab length

Commit:
Add basic styling site

Commit:
Rename index.js back to app.js

Commit:
Add Redux flow prototype and testing buttons

- No view components yet
- MapConstants.js exports three action types: TOGGLE_FACTION, TOGGLE_FILTER,
  SWITCH_ALL
- MapActions.js exports action creators for each type
- MapReducer.js creates the structure for the global store state and
  responds to dispatched actions
- configureStore.js creates the global store out of MapReducer.js, may be
  extended in the future with middleware
- SentriiApp is the root component for the entire app, it
  relies on react-redux's connect() to communicate with the store
- app.js renders the root container, SentriiApp, into a dom element on
  index.html, it also wraps the container in <Provider> to pass down the
  store as a prop so that it's available to connect() to

Commit:
Factor out test buttons into their own component

- Fixed some inconsistencies in spacing
- Added proptype validation to SentriiApp

Commit:
Add new mockup with better visual balance

Commit:
Add rudimentary rendering of component hierarchy

- Hierarchy structure still a prototype and not final
- No styles as of this commit
- The root controller-view/container was renamed to SentriiContainer
- Made semantical use of "switch" and "toggle" more consistent

Commit:
Update LICENSE with proper owner (me)

Commit:
Change compilation of gulpfile to ES6

Commit:
Add eslint as dev dependency

Commit:
Add rudimentary eslint config file

Commit:
Add rules and plugins to .eslintrc

Commit:
Fix longstanding tab inconsistency

Commit:
Insert end of file newline

Commit:
Fix all lint errors

Commit:
Remove trailing comma

Commit:
Add missing proptype validation

Commit:
Add styling for numerous elements

- Header, Footer, Filters Toggles are all styled
- Faction Switch needs to be styled
- Control Labels needs to be styled
- Added static placeholder for pixi canvas
- Created rudimentary grid system for layout based on a width of 1080px
  and 12 divisions
- Changed TOGGLE_ALL to TOGGLE_GROUP to be consistent with component names

Commit:
Add styling to faction switch

- Also hooked up the switch to its corresponding action
- Manipulating CSS classes seems to be the easiest way of achieving
  animations in React. I'm surprised I didn't think of it sooner.

Commit:
Fix reference to task CopyImages

Commit:
Darken switch handle highlight

Commit:
Add transparency to map image border

Commit:
Add comments to actions for detail

Commit:
Fix display bug on different zoom levels

Commit:
Change image aspect ratio for better borders

Commit:
Add offset to render switch handle correctly

Commit:
Normalize map borders and remove inset shadow

Commit:
Remove redundant backgrounds on filters

chikamichi commented:
I'd suggest you merge this branch and use a new one for the tests, for you're basically done with prototyping the app using Redux. What do you think?

chikamichi commented:
You have an open list of checkboxes though. Well, I'll see what you decide :+1:

kusera commented:
There's a little more I want to add to it before I merge it.

Commit:
Change object-fit of static map

Commit:
Add npm debug to ignore and remove .lnks

Commit:
Style control labels and hook to actions

Commit:
Shrink toggle group icons

Commit:
Hook up filter toggles

- Added styles to show active toggles
- Normalized names of filter groups across MapReducer and SentriiContainer
- Removed TestButton component and printState test function

Commit:
Add no-select to controls
